pkg/util/helper/status_test.go

```diff
@@ -0,0 +1,372 @@
+			key: types.NamespacedName{
+				Name:      "test-pod",
+				Namespace: "default",
+			},
+			obj: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-pod",
+					Namespace: "default",
+				},
+			},
+			mutateFn:      nil, // Will be set in the test
+			expectedError: "MutateFn cannot mutate object name and/or object namespace",
+		},
+		{
+			name: "attempt to mutate object namespace",
+			key: types.NamespacedName{
+				Name:      "test-pod",
+				Namespace: "default",
+			},
+			obj: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-pod",
+					Namespace: "default",
+				},
+			},
+			mutateFn:      nil, // Will be set in the test
+			expectedError: "MutateFn cannot mutate object name and/or object namespace",
+		},
+		{
+			name: "attempt to mutate both name and namespace",
+			key: types.NamespacedName{
+				Name:      "test-pod",
+				Namespace: "default",
+			},
+			obj: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-pod",
+					Namespace: "default",
+				},
+			},
+			mutateFn:      nil, // Will be set in the test
+			expectedError: "MutateFn cannot mutate object name and/or object namespace",
+		},
+		{
+			name: "successful mutation with labels and annotations",
+			key: types.NamespacedName{
+				Name:      "test-pod",
+				Namespace: "default",
+			},
+			obj: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-pod",
+					Namespace: "default",
+				},
+			},
+			mutateFn:      nil, // Will be set in the test
+			expectedError: "",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create local copy of the test object
+			testObj := tt.obj.DeepCopy()
+
+			// Set up mutation functions that operate on the local object
+			switch tt.name {
+			case "attempt to mutate object name":
+				tt.mutateFn = func() error {
+					testObj.SetName("modified-pod")
+					return nil
+				}
+			case "attempt to mutate object namespace":
+				tt.mutateFn = func() error {
+					testObj.SetNamespace("new-namespace")
+					return nil
+				}
+			case "attempt to mutate both name and namespace":
+				tt.mutateFn = func() error {
+					testObj.SetName("modified-pod")
+					testObj.SetNamespace("new-namespace")
+					return nil
+				}
+			case "successful mutation with labels and annotations":
+				tt.mutateFn = func() error {
+					testObj.SetLabels(map[string]string{"key": "value"})
+					testObj.SetAnnotations(map[string]string{"note": "test"})
+					return nil
+				}
+			}
+
+			err := mutate(tt.mutateFn, tt.key, testObj)
+
+			if tt.expectedError != "" {
+				assert.Error(t, err)
+				assert.Contains(t, err.Error(), tt.expectedError)
+				return
+			}
+			assert.NoError(t, err)
+
+			// Verify the object key hasn't changed for successful mutations
+			newKey := types.NamespacedName{
+				Name:      testObj.GetName(),
+				Namespace: testObj.GetNamespace(),
+			}
+
+			if tt.expectedError == "" {
```

Comment:
```suggestion
```

Lines `L322-L327` have already handled the case where `expectedError` is not empty, so there is no need to add a condition check for `tt.expectedError == ""`.